bin/ch-dir2squash

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+set -e
+
+libexec="$(cd "$(dirname "$0")" && pwd)"
+. "${libexec}/base.sh"
+
+# shellcheck disable=SC2034
+usage=$(cat <<EOF
+Pack a filesystem directory into a SquashFS file.
+
+Usage:
+
+  $ $(basename "$0") INDIR OUTDIR [ARGS ...]
+
+ARGS are passed unchanged to "mksquashfs".
+EOF
+)
+
+sentinel=WEIRD_AL_YANKOVIC
+
+parse_basic_args "$@"
+
+if [ $# -lt 2 ]; then
+    usage
+fi
+
+indir=$1
+outdir=$2
+shift 2
+
+# Check input directory exists
+if [ ! -e "$indir" ]; then
+    echo "can't squash: ${indir} does not exist" 1>&2
+    exit 1
+fi
+# Check input directory is a directory
+if [ ! -d "$indir" ]; then
+    echo "can't squash: ${indir} is not a directory" 1>&2
+    exit 1
+fi
+
+# Check OUTDIR exists and is a directory
+if [ ! -e "$outdir" ]; then
+    echo "can't squash: ${outdir} does not exist" 1>&2
+    exit 1
+fi
+if [ ! -d "$outdir" ]; then
+    echo "can't squash: ${outdir} is not a directory" 1>&2
+    exit 1
+fi
+
+# Ensure directories that ch-run needs exist.
```

Comment:
This was copied from ch-tar2dir which adds these files at the end of unpacking. But, I do see how the situations are not the same. I think using multiple directories or appending the extra things is possible. I'll look into making that work.

Reply:
Attempting to append those directories does not appear to work. If the directory is empty mksquashfs ignores it, and will not append it to the filesystem. The tar workflow already adds these directories, but in the case someone isn't using charliecloud tools adding these directories may be required. In the case of adding something below the empty directory it only appends the file to the root of the squashfs.

Reply:
Ok. Adding directories to a temp directory then appending them proves problematic. If the directories we are trying to add already exist we end up with copies. So if we are add `/dev`, but that exists in the image we end up with `/dev/` and `/dev_1/`. Which isn't great, but also not harmful as far as I know. 

We can't squash them at the same time or we end up with a squash holding two directories, the temp directory and the image tree. 

Finally, we could copy the directory the user provided into a temp directory, then add the things we need and squash the temp directory instead.

Reply:
The script now checks for everything in the input directory. If it exists it will do nothing. If something is missing it will create the needed things in a temp directory, and append them to the squash.